woodwork/table_accessor.py

```diff
@@ -1081,6 +1082,47 @@ def value_counts(self, ascending=False, top_n=10, dropna=False):
         """
         return _get_value_counts(self._dataframe, ascending, top_n, dropna)
 
+    @_check_table_schema
+    def infer_temporal_frequencies(self, temporal_columns=None):
+        """Infers the observation frequency (daily, biweekly, yearly, etc) of each temporal column
+            in the DataFrame. Temporal columns are ones with the logical type Datetime or Timedelta.
+
+        Args:
+            temporal_columns (list[str], optional): Columns for which frequencies should be inferred. Must be columns
+                that are present in the DataFrame and are temporal in nature. Defaults to None. If not
+                specified, all temporal columns will have their frequencies inferred.
+
+        Returns:
+            (dict): A dictionary where each key is a temporal column from the DataFrame, and the
+                value is its observation frequency represented as a pandas offset alias string (D, M, Y, etc.)
+                or None if no uniform frequency was present in the data.
+
+        Note:
```

Comment:
@freddyaboulton @thehomebrewnerd Tried to cover all of the behaviors that I could find for `infer_freq` in the bullet points below. Let me know if there's anything you thing I should add or if anything is confusingly written!

Reply:
This is awesome! Thanks for adding this.